packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -608,6 +609,45 @@ class NavigatorObserver {
   void didStopUserGesture() { }
 }
 
+/// An inherited widget to host a hero controller.
+///
+/// This class should not be used directly. The [MaterialApp] and [CupertinoApp]
+/// use this class to host the [HeroController], and they should be the only
+/// exception to use this class. If you want to subscribe your own
+/// [HeroController], use the [Navigator.observers] instead.
```

Comment:
why? it seems perfectly reasonable to use this class to provide a HeroController if you're making a widget that gets a Navigator from somewhere else.

Reply:
I think we should encourage user to use observer instead if they can?

Reply:
I mean until we finally refactor the hero controller to not depend on navigator observer